benchmark/utils/DriverUtils.swift

```diff
@@ -380,16 +363,16 @@ func runBench(_ test: Test, _ c: TestConfig) -> BenchResults? {
     print("Running \(test.name) for \(c.numSamples) samples.")
   }
 
+  test.setUpFunction?()
+
   let sampler = SampleRunner()
   for s in 0..<c.numSamples {
     let time_per_sample: UInt64 = 1_000_000_000 * UInt64(c.iterationScale)
 
     var scale : UInt
     var elapsed_time : UInt64 = 0
     if c.fixedNumIters == 0 {
-      test.setUpFunction?()
       elapsed_time = sampler.run(test.name, fn: testFn, num_iters: 1)
-      test.tearDownFunction?()
```

Comment:
Please do not move this out of the loop. It is important that we have setup/teardown before/after each invocation of the benchmarking function. Benchmark writers should be able to assume that they will have a pristine state when the benchmark starts running. Otherwise, you could have one benchmark iteration affecting another benchmark iteration.

Reply:
I believe this refactoring is safe, because multiple iterations are handled by passing the scale parameter to the `sampler.run` method, see [line 402](https://github.com/palimondo/swift/blob/4fe861ae5890467b76813ad4b0448507b74711e0/benchmark/utils/DriverUtils.swift#L402). This means that `setUp` and `tearDown` aren't invoked around every iteration. 

Even if you're thinking about a case which measures some kind of a mutable operation (say removing elements from collection), the `run` function already has to handle the setup of the workload for each iteration. In such a case you'd do the expensive collection building in the `setUp`, and do a cheaper copy at the beginning of each inner loop, releasing the collection in `tearDown`. All this implies that the proper names for these functions should be `beforeTest` and `afterTest`, but that's for another refactoring.

That's why I think that measuring multiple samples with single `setUp` and `tearDown` hoisted outside of the sample loop is safe.

Reply:
I agree, the setUp/tearDown should be inside the loop

Reply:
@eeckstein This timing of our comments look simultaneous. Did you read my explanation?